protos/service.proto

```diff
@@ -0,0 +1,39 @@
+syntax = "proto3";
+
+import "google/protobuf/empty.proto";
+
+package mapreduce;
+
+
+service MapReduce {
+
+    rpc WatchHealth(google.protobuf.Empty) returns (stream google.protobuf.Empty) {}
+    rpc Map(MapTask) returns (MapResponse) {}
+    rpc Reduce(ReduceTask) returns (ReduceResponse) {}
+    rpc ReduceChangePath(ReduceChangePathMes) returns (google.protobuf.Empty) {}
+
+}
+
+message MapTask {
+    string file_path = 1;
+    int32 num_partitions = 2;
+}
+
+message MapResponse {
+    repeated string partition_paths = 1;
+}
+
+message ReduceTask {
```

Comment:
[question]  do we assume here that `partition_paths` contains `num_mappers` paths?
- if so - why do we need to pass `num_mappers` + in this case I guess we can only start reducing after all mappers are done, right?
- if not (so I'm assuming we send the paths in some chunks and count if we received a path for each existing map task and that's why we need `num_mappers`) - maybe we could utilize a streaming rpc instead to receive the paths

Reply:
just noticed that I could've marked multiple lines at once, will do that in next reviews

Reply:
Removed num_mappers, for now let's assume reducers start after mappers, we can try relaxing it later